fdbserver/masterserver.actor.cpp

```diff
@@ -998,6 +1004,36 @@ ACTOR Future<Void> provideVersions(Reference<MasterData> self) {
 	}
 }
 
+ACTOR Future<Void> serveLiveCommittedVersion(Reference<MasterData> self) {
+	loop {
+		choose {
+			when(GetRawCommittedVersionRequest req = waitNext(self->myInterface.getLiveCommittedVersion.getFuture())) {
+				Span span("MS:getLiveCommittedVersion"_loc, { req.spanContext });
+				if (req.debugID.present())
+					g_traceBatch.addEvent("TransactionDebug", req.debugID.get().first(), "MasterServer.serveLiveCommittedVersion.GetRawCommittedVersion");
+
+				if (self->liveCommittedVersion == invalidVersion) {
```

Comment:
ASSERT(self->lifeCommittedVersion != invalidVersion);

Reply:
Well, after second thought, the recovery transaction needs to read a live committed version from master as well and at that time it is equal to 'invalidVersion'.

Reply:
After we initialize lifeCommittedVersion to be invalidVersion, we didn't update it to be recoveryTransactionVersion when we finish the transaction, so he it can still be invalidVersion. We prefer setting it to be recoveryTransactionVersion here over setting it after the recovery transaction because it is easy to reason this way.

Reply:
Maybe add a one line comment in the code to explain it? If the code can confuse Evan, it is definitely confusing. ;)